### PR TITLE
Rework file provider for testing

### PR DESCRIPTION
**Pull request by SavikPavel**
### Description

FileProvider rework and tests added

### API Changes
Whole IFileProvider interface reworked

### Platforms Affected

- Core (all platforms)

### Behavioral/Visual Changes

None

### Before/After Screenshots

Not applicable

### PR Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/Softeq/XToolkit.WhiteLabel/blob/master/.github/CONTRIBUTING.md) document
- [x] My code follows the [code styles](https://github.com/Softeq/dotnet-guidelines)
- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Softeq.XToolkit.Common/Softeq.XToolkit.Common.csproj [hunk 1]
```diff
@@ -5,4 +5,7 @@
     <TargetFramework>netstandard2.1</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="System.IO.Abstractions" Version="12.1.1" />
```
Comment (wcoder) [inline Softeq.XToolkit.Common/Softeq.XToolkit.Common.csproj:9]: brakes the idea of ​​common library - "no dependencies"

File: Softeq.XToolkit.Common.Tests/Files/BaseFileProviderTests/BaseFileProviderTests.cs [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+        }
+
+        [Theory]
+        [InlineData(null, "content")]
+        [InlineData("", "content")]
+        public async void WriteFileAsync_InvalidPath_ThrowsException(
+            string path, string content)
+        {
+            byte[] byteArray = Encoding.ASCII.GetBytes(content);
+            MemoryStream stream = new MemoryStream(byteArray);
```
Comment (wcoder) [inline Softeq.XToolkit.Common.Tests/Files/BaseFileProviderTests/BaseFileProviderTests.cs:312]: can be extracted to separate method: `CreateStream(string)`

Comment (SavikPavel) [reply Softeq.XToolkit.Common.Tests/Files/BaseFileProviderTests/BaseFileProviderTests.cs]: Done

File: Softeq.XToolkit.Common/Files/IFileProvider.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        Task<Stream> GetFileContentAsync(string path);
+
+        /// <summary>
+        ///     Copies file asynchronously to the new path.
+        /// </summary>
+        /// <param name="srcPath">Source file path.</param>
+        /// <param name="dstPath">Destination file path.</param>
+        /// <param name="overwrite">Overwrite destination file if exists.</param>
+        /// <returns>Copy file Task.</returns>
+        Task CopyFileAsync(string srcPath, string dstPath, bool overwrite);
```
[the remark this answers is not in the document]
Comment (SavikPavel) [reply Softeq.XToolkit.Common/Files/IFileProvider.cs]: Renamed

**Review by AlexandraMash — APPROVED** [review]

**Comment by wcoder** [comment]
@SavikPavel Could you please fix the build error?
```
ImagePicker/ImagePicker.cs(77,44): error CS1061: 'IFileProvider' does not contain a definition for 'WriteStreamAsync' and no accessible extension method 'WriteStreamAsync' accepting a first argument of type 'IFileProvider' could be found (are you missing a using directive or an assembly reference?)
```

**Comment by wcoder** [comment]
Is [this issue](https://github.com/System-IO-Abstractions/System.IO.Abstractions/issues/634) affect us?

**Comment by wcoder** [comment]
```
"/Users/runner/work/1/s/samples/Playground/Playground.iOS/Playground.iOS.csproj" (default target) (4:6) ->
(_CompileToNative target) -> 
  MTOUCH : error MT2101: Can't resolve the reference 'System.Void System.IO.FileSystemAclExtensions::Create(System.IO.DirectoryInfo,System.Security.AccessControl.DirectorySecurity)', referenced from the method 'System.Void System.IO.Abstractions.DirectoryInfoWrapper::Create(System.Security.AccessControl.DirectorySecurity)' in 'System.IO.FileSystem.AccessControl, Version=4.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [/Users/runner/work/1/s/samples/Playground/Playground.iOS/Playground.iOS.csproj]


```

**Review by wcoder — APPROVED** [review]

**Pull request merged**